airflow/dag_processing/bundles/git.py

```diff
@@ -148,8 +148,8 @@ def __init__(
         self.git_conn_id = git_conn_id
         self.repo_url = repo_url
         try:
-            self.hook = GitHook(git_conn_id=self.git_conn_id)
-            self.repo_url = self.repo_url or self.hook.repo_url
+            self.hook = GitHook(git_conn_id=self.git_conn_id, repo_url=self.repo_url)
+            self.repo_url = self.hook.repo_url
         except AirflowException as e:
```

Comment:
Not related to this PR, but should we make it a more concrete exception instead of the broad AirflowException 🤔

Reply:
Maybe. The most obvious failure mode is missing connection. But I wonder if there are others... I'll look into this tomorrow, I've taken a note.